metricbeat/module/logstash/node_stats/node_stats.go

```diff
@@ -107,3 +97,23 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 
 	return nil
 }
+
+func (m *MetricSet) init() error {
+	if m.XPack {
+		logstashVersion, err := logstash.GetVersion(m.MetricSet)
+		if err != nil {
+			return err
+		}
+
+		arePipelineGraphAPIsAvailable := logstash.ArePipelineGraphAPIsAvailable(logstashVersion)
+
+		if !arePipelineGraphAPIsAvailable {
+			const errorMsg = "the %v metricset with X-Pack enabled is only supported with Logstash >= %v. You are currently running Logstash %v"
+			return fmt.Errorf(errorMsg, m.FullyQualifiedName(), logstash.PipelineGraphAPIsAvailableVersion, logstashVersion)
+		}
+
+		m.HTTP.SetURI(m.HTTP.GetURI() + "?vertices=true")
+	}
+
```

Comment:
@ycombinator The two init function looks really close to me could it be possible to move them to the logstash package and use the same?

Reply:
@ph I extracted much of the common code into the `logstash` package. Please take a look at let me know what you think. Thanks!